src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -847,23 +847,34 @@ def is_frontend_port_present_in_host():
 
 
 def get_num_dpus():
-    # Todo: we should use platform api to get the dpu number
-    # instead of rely on the platform env config.
-    num_dpus = 0
-    platform_env_conf_file_path = get_platform_env_conf_file_path()
+    """
+    Retrieves the number of DPUs from platform.json file.
 
-    # platform_env.conf file not present for platform
-    if platform_env_conf_file_path is None:
-        return num_dpus
+    Args:
 
-    # Else open the file check for keyword - num_dpu -
-    with open(platform_env_conf_file_path) as platform_env_conf_file:
-        for line in platform_env_conf_file:
-            tokens = line.split('=')
-            if len(tokens) < 2:
-               continue
-            if tokens[0].lower() == 'num_dpu':
-                num_dpus = tokens[1].strip()
-                break
-    return int(num_dpus)
+    Returns:
+        A integer to indicate the number of DPUs.
+    """
+
+    platform = get_platform()
+    if not platform:
+        return 0
+
+    (platform_path, hwsku_path) = get_paths_to_platform_and_hwsku_dirs()
+
+    # Check for 'hwsku.json' file presence first
+    hwsku_json_file = os.path.join(hwsku_path, HWSKU_JSON_FILE)
+
+    if os.path.isfile(hwsku_json_file):
```

Comment:
@xincunli-sonic Why do we check for hwsku?

Reply:
Fixed, I did not notice there is a dedicated API can retrieve platform file directly.